pylint/checkers/imports.py

```diff
@@ -914,11 +914,27 @@ def _add_imported_module(self, node: ImportNode, importedmodname: str) -> None:
 
     def _check_preferred_module(self, node: ImportNode, mod_path: str) -> None:
         """Check if the module has a preferred replacement."""
-        if mod_path in self.preferred_modules:
+
+        mod_compare = [mod_path]
+        # build a comparison list of possible names using importfrom
+        if isinstance(node, astroid.nodes.node_classes.ImportFrom):
+            mod_compare = [f"{node.modname}.{name[0]}" for name in node.names]
+
+        # find whether there are matches with the import vs preferred_modules keys
+        matches = [
+            key
+            for key in list(self.preferred_modules.keys())
+            for compare_val in mod_compare
+            # determine if key is in compare_val
+            if key in compare_val
+        ]
+
+        # if we have matches, add message
+        if len(matches) > 0:
```

Comment:
```suggestion
        if matches:
```
We have a check for that, it must be a false negative.

Reply:
Thank you! I'll get this updated.